llama-index-integrations/llms/llama-index-llms-ollama/llama_index/llms/ollama/base.py

```diff
@@ -107,329 +167,239 @@ def _model_kwargs(self) -> Dict[str, Any]:
             **self.additional_kwargs,
         }
 
+    def _convert_to_ollama_messages(self, messages: Sequence[ChatMessage]) -> Dict:
+        return [
+            {
+                "role": message.role.value,
+                "content": message.content or "",
+            }
+            for message in messages
+        ]
+
+    def _prepare_chat_with_tools(
+        self,
+        tools: List["BaseTool"],
+        user_msg: Optional[Union[str, ChatMessage]] = None,
+        chat_history: Optional[List[ChatMessage]] = None,
+        verbose: bool = False,
+        allow_parallel_tool_calls: bool = False,
+        **kwargs: Any,
+    ) -> Dict[str, Any]:
+        tool_specs = [
+            tool.metadata.to_openai_tool(skip_length_check=True) for tool in tools
+        ]
+
+        if isinstance(user_msg, str):
+            user_msg = ChatMessage(role=MessageRole.USER, content=user_msg)
+
+        messages = chat_history or []
+        if user_msg:
+            messages.append(user_msg)
+
+        return {
+            "messages": messages,
+            "tools": tool_specs or None,
+        }
+
+    def _validate_chat_with_tools_response(
```

Comment:
lol i want to refactor this so we don't have to define for every LLM, but anyhow